Providers/Scripts/2.4x-2.5x/Scripts/Tests/test_nxProviders.py

```diff
@@ -717,17 +665,15 @@ def tearDown(self):
         """
         time.sleep(4)
```

Comment:
Are all the sleeps necessary? By having two sleeps in setup and teardown each test takes at least 14 seconds! Unit tests usually take milliseconds. Have you considered separating unit tests from system tests?